src/main/java/org/broadinstitute/consent/http/service/ElectionService.java

```diff
@@ -184,11 +188,20 @@ public Election updateElectionById(Election rec, Integer electionId) {
         return electionDAO.findElectionWithFinalVoteById(electionId);
     }
 
-    public Election submitFinalAccessVoteDataRequestElection(Integer electionId) throws Exception {
+    public Election submitFinalAccessVoteDataRequestElection(Integer electionId, String darId) throws Exception {
         Election election = electionDAO.findElectionWithFinalVoteById(electionId);
         if (election == null) {
             throw new NotFoundException("Election for specified id does not exist");
         }
+        DataAccessRequest dar = dataAccessRequestService.findByReferenceId(darId);
+        if (dar == null) {
+            throw new NotFoundException("Data Access Request for specified id does not exist");
+        } 
+        Integer userId = dar.getUserId();
+        List<LibraryCard> libraryCards = Objects.nonNull(userId) ? libraryCardDAO.findLibraryCardsByUserId(userId) : new ArrayList<>();
+        if (libraryCards.isEmpty()) {
+            throw new NotFoundException("No Library cards exist for the researcher on this dar");
+        }
```

Comment:
This introduces two new failure conditions that we should describe in `api-docs.yaml`. That only covers a missing election so we should add the cases that the DAR or the Library Card might not be found either.